src/commands/mod.rs

```diff
@@ -90,19 +90,14 @@ impl Cli {
         }
     }
 
-    pub fn run(self, py: pyo3::Python<'_>) -> pyo3::PyResult<()> {
+    pub fn run(self, py: pyo3::Python<'_>) -> crate::error::Result<()> {
         let global = &self.global;
         sentry::configure_scope(|scope| {
             scope.set_extra("python.version", py.version().into());
             scope.set_extra("aqora.url", global.url.clone().into());
         });
 
-        pyo3_asyncio::tokio::run::<_, ()>(py, async move {
-            if let Err(e) = self.do_run().await {
-                tracing::error!("{}", e);
-                std::process::exit(1)
-            }
-            std::process::exit(0);
-        })
+        let result = pyo3_asyncio::tokio::run(py, async move { Ok(self.do_run().await) });
```

Comment:
finalizers do not get run on `std::process:exit()`

Reply:
ah okay, so this is to send errors to sentry on exit? i think we only want to send system errors, so it would probably make more sense to capture them here, filter and exit gracefully or?

Reply:
There are three finalizers in this PR that ran: sentry, tracing, and GC. But whatever they are, I think it is easier to reason about the program without this call

Reply:
makes sense! that change to `run` should handle all of my concerns about backtraces and what not. you could also return `ExitCode::SUCCESS` and `ExitCode::FAILURE` here instead of process exit to return the proper exit code without throwing the ugly backtrace i think